api/python/quilt3/packages.py

```diff
@@ -1074,7 +1084,26 @@ def dump(self, writable_file):
         return self._dump(writable_file)
 
     def _dump(self, writable_file):
-        writer = jsonlines.Writer(writable_file)
+        json_encode = json.JSONEncoder(ensure_ascii=False).encode
+
+        def dumps(obj):
+            data = json_encode(obj)
+            encoded_size = len(data.encode())
+            if encoded_size > MANIFEST_MAX_RECORD_SIZE:
+                lk = obj.get('logical_key')
+                entry_text = 'package metadata' if lk is None else f'entry with logical key {lk!r}'
+                raise QuiltException(
+                    f"Size of manifest record for {entry_text} is {encoded_size} bytes, "
+                    f"but it's limited by {MANIFEST_MAX_RECORD_SIZE} bytes. "
```

Comment:
```suggestion
                    f"but should be less than {MANIFEST_MAX_RECORD_SIZE} bytes. "
```

Reply:
"should" would make sense if it was a warning, but it's an exception. Use "must" instead?

Reply:
Yeah, fixed